keewe-api/src/main/java/ccc/keeweapi/service/challenge/ChallengeApiService.java

```diff
@@ -38,4 +42,19 @@ public ParticipationCheckResponse checkParticipation() {
         boolean participation = challengeDomainService.checkParticipation(SecurityUtil.getUserId());
         return challengeAssembler.toParticipationCheckResponse(participation);
     }
+
+    @Transactional(readOnly = true)
+    public InsightProgressResponse getMyParticipationProgress() {
+        Long userId = SecurityUtil.getUserId();
+        ChallengeParticipation participation =
+                challengeDomainService.findCurrentParticipationWithChallenge(userId).orElse(null);
+
+        if (Objects.isNull(participation)) {
+            return null;
+        }
+
+        Long current = insightDomainService.getRecordedInsightNumber(participation);
+
+        return challengeAssembler.toParticipationProgressResponse(participation, current);
+    }
```

Comment:
nit: functional programming을 해봐도 좋을거같아요 ㅎ

```suggestion
        return challengeDomainService.findCurrentParticipationWithChallenge(SecurityUtil.getUserId())
                .stream()
                .filter(Objects::nonNull)
                .map(challengeParticipation -> Pair.of(challengeParticipation, insightDomainService.getRecordedInsightNumber(challengeParticipation)))
                .map(pair -> challengeAssembler.toParticipationProgressResponse(pair.getFirst(), pair.getSecond()))
                .findFirst()
                .orElse(null);
    }
```

Reply:
```suggestion
          Long userId = SecurityUtil.getUserId();

        return challengeDomainService.findCurrentParticipationWithChallenge(userId)
                .map(this::toParticipationProgressResponse)
                .orElse(null);
    }

    private InsightProgressResponse toParticipationProgressResponse(ChallengeParticipation participation) {
        Long current = insightDomainService.getRecordedInsightNumber(participation);
        return challengeAssembler.toParticipationProgressResponse(participation, current);
    }
```

Reply:
```suggestion
        Long userId = SecurityUtil.getUserId();

        return challengeDomainService.findCurrentParticipationWithChallenge(userId)
                .map(participation -> {
                    Long current = insightDomainService.getRecordedInsightNumber(participation);
                    return challengeAssembler.toParticipationProgressResponse(participation, current);
                })
                .orElse(null);
    }
```

Reply:
Optional에서 map 된다는걸 이제 알았네요..
코드 참고해서 수정해봤어요. 혹시 두 방법 중에 어떤게 더 보기 편한가요?

Reply:
아래가 나은거같아요~

Reply:
감사합니다 ㅎㅎ 하나 배워가네요🙏